### PR TITLE
Fix and improve Spray JSON docs

### PR DESCRIPTION
**Pull request by jonas**
- CompactPrinter is now the default so change the JSON printer example to mention PrettyPrinter
- Replace project version in the SBT snippet.
- Reformat Markdown

NOTE: the last point is something we might want to raise upstream in Paradox, to see if we can make it optional whether line breaks are honours in the Markdown rendering. Alternatively, we can reformat all docs to not use new lines inside paragraphs.

### CODE REVIEW

File: docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala [hunk 1]
```diff
@@ -7,30 +7,30 @@ import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
...
+    class MyJsonService extends Directives with PrettyJsonFormatSupport {
 
       // format: OFF
       val route =
         get {
           pathSingleSlash {
             complete {
-              // should complete with spray.json.JsValue = {"name":"akka","id":42}
-              CompactPrintedItem("akka", 42) // will render as JSON
+              // should complete with spray.json.JsValue = { "name": "akka", "id": 42 }
```
Comment (jonas) [inline docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala]: Note, I didn't actually test this.

File: docs/src/main/paradox/scala/http/common/json-support.md [hunk 2]
```diff
@@ -1,37 +1,47 @@
...
- 1. Add a library dependency onto `"com.typesafe.akka" %% "akka-http-spray-json-experimental" % "@version@"`. <!-- FIXME: use preprocessing -->
- 2. `import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._` or mix in the
-`akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport` trait.
- 3. Provide a `RootJsonFormat[T]` for your type and bring it into scope.
-Check out the [spray-json](https://github.com/spray/spray-json) documentation for more info on how to do this.
+@@@vars
+```sbt
+"com.typesafe.akka" %% "akka-http-spray-json-experimental" % "$project.version$"`
+```
+@@@
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/common/json-support.md]: 👍

File: docs/src/main/paradox/scala/http/common/json-support.md [hunk 3]
```diff
@@ -1,37 +1,47 @@
...
+
+Next, provide a `RootJsonFormat[T]` for your type and bring it into scope. Check out the [spray-json] documentation for more info on how to do this.
+
+Finally, mix in the `akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport` trait as shown in the example below or import the `FromEntityUnmarshaller[T]` and `ToEntityMarshaller[T]` implicits directly from `SprayJsonSupport`
+
+```scala
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+```
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/common/json-support.md]: good

File: docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala (hunk 1 above)
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala]: AFAIR pretty is:

```
{ 
  "name": "akka",
  "id": 42
}
```

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
@akka/akka-http-team, @jrudolph please feel free to LGTM and merge once PR hits 2 LGTMs (anyone within http-team is welcome to do so) :-)

File: docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala (hunk 1 above)
Comment (2beaucoup) [inline docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala]: You can add
```scala
Get("/") ~> route ~> check {
  responseAs[String] shouldEqual """..."""
}
```

Comment (jonas) [inline docs/src/test/scala/docs/http/scaladsl/SprayJsonPrettyMarshalSpec.scala]: Awesome, added a test to the example.

**Comment by jonas** [comment]
The failure looks unrelated:

```
[info]   - the config setting (chunked entity) *** FAILED ***
[info]     null did not equal EntityStreamSizeException: actual entity size (None) exceeded content length limit (10 bytes)! You can configure this by setting `akka.http.[server|client].parsing.max-content-length` or calling `HttpEntity.withSizeLimit` before materializing the dataBytes stream.,
[info]     inside HttpResponse(200 OK,List(),HttpEntity.Chunked(application/octet-stream),HttpProtocol(HTTP/1.1)) (LowLevelOutgoingConnectionSpec.scala:564)
```

**Comment by jypma** [comment]
Mmm, we just had the exact same test failure over at https://github.com/akka/akka-http/pull/323. Coincidence? I think NOT! :)

**Comment by jonas** [comment]
Is there a way to ask @akka-ci to rebuild?

**Comment by ktoso** [comment]
PLS BUILD

**Comment by jonas** [comment]
Thanks. Now it is failing with the error mentioned in #298:

```
[info]   - a 200 response to a HEAD request *** FAILED ***
[info]     java.util.concurrent.TimeoutException: Futures timed out after [500 milliseconds]
[info]     at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]     at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
[info]     at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:190)
[info]     at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]     at scala.concurrent.Await$.result(package.scala:190)
[info]     at akka.http.impl.engine.parsing.ResponseParserSpec$Test.collectBlocking(ResponseParserSpec.scala:321)
```

**Comment by ktoso** [comment]
That's too tight timeouts IMO,

**Comment by jonas** [comment]
Rebased to see if this will trigger a rebuild now that jenkins.akka.io seems more quiet.

**Comment by jonas** [comment]
🎱

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**